pandas/core/indexes/multi.py

```diff
@@ -1351,13 +1351,14 @@ def __contains__(self, key: Any) -> bool:
     def dtype(self) -> np.dtype:
         return np.dtype("O")
 
+    @cache_readonly
```

Comment:
I'm thinking if we're going to cache a method, we should do so to all super/sub classes unless there is a good reason not to. Especially in this case because it's just a Boolean value. Any reason not to?

Reply:
Sure the symmetry argument make sense to me. I'll add `cache_readonly` to the base method too